src/main/java/com/github/fge/jsonpatch/RegistryBasedJsonPatchFactory.java

```diff
@@ -0,0 +1,89 @@
+package com.github.fge.jsonpatch;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.github.fge.jackson.JacksonUtils;
+import com.github.fge.msgsimple.bundle.MessageBundle;
+import com.github.fge.msgsimple.load.MessageBundles;
+
+import java.io.IOException;
+import java.util.*;
+
+/**
+ * RegistryBasedJsonPatchFactory is responsible for parsing JsonNodes into JsonPatch objects based on
+ * which JsonPatchOperations it knows about in its registry.
+ */
+public class RegistryBasedJsonPatchFactory implements JsonPatchFactory
+{
+    private Map<String, Class<? extends JsonPatchOperation>> operations;
+
+    private static final MessageBundle BUNDLE
+        = MessageBundles.getBundle(JsonPatchMessages.class);
+
+    /**
+     * Builds a JsonPatchFactory based on the patch operations we want to register
+     */
+    public static class RegistryBasedJsonPatchFactoryBuilder {
```

Comment:
I like the Builder name.  I put it inside the class because I wanted a private constructor for the actual Factory, and force construction through the Builder.